### ISSUE TITLE
 webhooks working?

### ISSUE THREAD

**Issue by Sz4by** [Carbowix/newswire#9]
Hmm?

**Comment by Sz4by** [comment]
Where should I enter?

**Comment by Carbowix** [comment]
Hi @Sz4by ,
The installation is pretty simple.
- Extract the files into a folder and run `npm install` through your CMD. 
- Create a file in the same folder maybe called `index.js`
- Add the following code to it:
```js
 const newswire = require('./newswire');
 const latestNews = new newswire(type, webhookURL);
```
- Replace "type" with any of the following types: `rdr2`, `gtav`, `latest`, `music`, `fanart`, `fanvideos`, `creator`, `tips`, `rockstar`, `updates`.
- Replace webhookURL with your [webhook's URL](https://support.discordapp.com/hc/en-us/articles/228383668-Intro-to-Webhooks).

Example of using `gtav` as the type
```js
 const newswire = require('./newswire');
 const latestNews = new newswire("gtav", "https://...");
```
- Save the file and run `node index.js` in your CMD. It should show the news every 2 hours if there's any new articles on the newswire.

Hopefully, I'll organize the project to make it more user-friendly, but that should do it for now. If you have any issues, feel free to reply to this issue or open a new one.

**Comment by Sz4by** [comment]
And it doesn't matter where I enter it?

**Comment by Carbowix** [comment]
> And it doesn't matter where I enter it?

Yeah as long the library is linked correctly, you can put that piece of code anywhere, and it should run. Whether it is a bot or another script that runs all the time.

**Comment by Sz4by** [comment]
Thank you, and one more question, it's not possible to put Hungarian language, right?

**Comment by Carbowix** [comment]
> Thank you, and one more question, it's not possible to put Hungarian language, right?

Library currently doesn't support language localization for news, but I'll probably give it a small update later this week to support languages and other stuff.

For now, you can change the following [line](https://github.com/Carbowix/rockstar-newswire/blob/master/newswire.js#L140) with your language locale code. For example I want it to be German, so I would change it to `"de_de"`

But for Hungarian language `"hu_hu"` there's bad news, Rockstar Newswire doesn't support Hungarian locale. It only supports the following languages. (Maybe try it, who knows it might work)
![image](https://github.com/user-attachments/assets/3cd656c0-9445-4bcc-b730-9c8a737b4189)

**Comment by Sz4by** [comment]
It's ok thanks

**Comment by Sz4by** [comment]
Is it okay if I put it at the beginning?
![Screenshot_20241027-123820~2](https://github.com/user-attachments/assets/e784ca1e-4356-4ee4-8d1e-7a723ccc24f5)

**Comment by Carbowix** [comment]
if you'll put it on the same file as the `newswire.js` then you won't need the first line and to run it would be `node newswire.js`

**Comment by Sz4by** [comment]
[READY] Started news feed for gtav. Feed refreshes every 2 hours.
This is what he says to me, but he hasn't sent anything to discord yet

**Comment by Carbowix** [comment]
> [READY] Started news feed for gtav. Feed refreshes every 2 hours.
> This is what he says to me, but he hasn't sent anything to discord yet

If it's just freshly deployed, it should send the latest article in the category, and it'll refresh every 2 hours in case a new article is released in that time-frame. If it didn't send, make sure that webhook url is correct

**Comment by Sz4by** [comment]
_I think it only works on Windows and not on hosting?_

**Comment by Sz4by** [comment]
`Isn't there one that works on Linux?`

**Comment by Carbowix** [comment]
Yeah I think, it won't work with Linux for now. At the time of its creation, I was just making it for fun. Probably I'll update it this week with quality-of-life changes.

**Comment by Sz4by** [comment]
It would be nice if Linux worked

**Comment by sharpie1664** [comment]
how would you use more then 1 tag would it be const latestNews = new newswire(gta5,rdr2, webhookURL);

**Comment by Carbowix** [comment]
> how would you use more then 1 tag would it be const latestNews = new newswire(gta5,rdr2, webhookURL);

hey @sharpie1664 
Currently it doesn't support more than 1 tag, but for now you can do multiple by creating multiple variables until I get the time to release the new update.

Either use `latest` to get all possible news or use the following method:
Ex: 
```js
const gtaNews = new newswire('gtav', webhookURL);
const rdrNews = new newswire('rdr2', webhookURL);
```

**Comment by DaniCariga** [comment]
Hello, after I am running the node index.js I get this error
```
    at wrapSafe (node:internal/modules/cjs/loader:1662:18)
    at Module._compile (node:internal/modules/cjs/loader:1704:20)
    at Object..js (node:internal/modules/cjs/loader:1895:10)
    at Module.load (node:internal/modules/cjs/loader:1465:32)
    at Function._load (node:internal/modules/cjs/loader:1282:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:171:5)
    at node:internal/main/run_main_module:36:49
```

I did something wrong ?

**Comment by Sz4by** [comment]
I managed to solve the problem, it works now, I connected the discord bot to it and managed to get it to run on Linux too 

![Image](https://github.com/user-attachments/assets/f057e231-10ea-4c2a-92c6-c6375464a2f5)
![Image](https://github.com/user-attachments/assets/b6aea296-d7a4-48b5-a946-a325071b4cae)

**Comment by NightMean** [comment]
@Sz4by if you are interested, I forked this project and added an option to run it via docker. 

https://github.com/NightMean/rockstar-newswire-tracker
